docker/Dockerfile

```diff
@@ -52,9 +52,9 @@ COPY pyproject.toml ./
 RUN --mount=type=cache,target=/root/.cache/pip \
     # xformers + triton fails to install on arm64
     if [ "$GPU_DRIVER" = "cuda" ] && [ "$TARGETPLATFORM" = "linux/amd64" ]; then \
-        pip install -e ".[xformers]"; \
+        pip install $extra_index_url_arg -e ".[xformers]"; \
```

Comment:
For Linux+CUDA, we shouldn't pass the `--extra-index-url` flag - the official Torch instructions guide us to install it from PyPi, and we've learned over time that deviating from those official instructions result in things randomly breaking. Great catch on it being unused in the `else` condition though.

Reply:
Also, I just noticed that on line 38 we're setting `--index-url` instead of `--extra-index-url` - this will probably break the install because `pip` will not look at PyPi anymore. This was introduced in some prior PR, but could I ask you to please fix it here? Thank you! :pray:

Reply:
In that case, might it make sense to just leave the argument empty entirely for the CUDA case? No matter what we do here, the argument is used to install torch in line 41, which would also deviate from torch's instructions as I understand you.

As for `--index-url` and `--extra-index-url`, I totally missed that. I already had a working build before I added $extra_index_url_arg to line 57, so it didn't break for me. I can imagine that someone wanted to make sure that torch really wouldn't be installed from PyPi? It is my understanding that `--extra-index-url` takes precedence, though, so changing it should be fine.